feat: add C implementation for `stats/base/dists/chi/pdf`

AadishJ opened the pull request:
Resolves #3495.

## Description

> What is the purpose of this pull request?

This pull request:

- Adds a C implementation of the chi pdf.
- Includes benchmarks for the C implementation.
- Provides examples demonstrating the usage of the chi pdf in C.


## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3495

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

- [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Commit:
feat(chi-pdf): adding c implementation

AadishJ commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

AadishJ committed:
Update README.md

Signed-off-by: Aadish Jain <jain.aadishj@gmail.com>

AadishJ committed:
Update manifest.json

Signed-off-by: Aadish Jain <jain.aadishj@gmail.com>

Planeshifter committed:
fix: correct header order, validation, style issues, and use uniform over randu

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/chi/pdf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,140 @@
+* limitations under the License.
+*/
+
+#include <sys/time.h>
+#include "stdlib/stats/base/dists/chi/pdf.h"
```
```suggestion
#include "stdlib/stats/base/dists/chi/pdf.h"
#include <sys/time.h>
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/chi/pdf/examples/c/example.c:
```diff
@@ -0,0 +1,41 @@
+		y = stdlib_base_dists_chi_pdf( x, k );
+		printf( "x: %lf, k: %lf, f(x;k): %lf\n", x, k, y );
+	}
+	return 0;
```
```suggestion
```

Planeshifter committed:
fix: chi pdf c implementation and tests

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
chore: minor clean-up

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
Merge branch 'develop' into develop14

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/chi/pdf/benchmark/benchmark.js:
```diff
@@ -73,6 +73,10 @@ bench( pkg+':factory', function benchmark( b ) {
 
 	k = 10.0;
 	mypdf = pdf.factory( k );
+	x = new Float64Array( 100 );
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = uniform( -2.0, 2.0 );
+	}
```
```suggestion
```

Planeshifter committed:
chore: remove duplicated logic

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Should be ready to land!

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/chi/pdf/test/test.native.js:
```diff
@@ -0,0 +1,121 @@
+		} else {
+			delta = abs( y - expected[ i ] );
+			tol = 2.0 * EPS * abs( expected[ i ] );
+			if ( tol < 1e-14 ) {
```
@Planeshifter This should have been flagged. This is not how we do tolerances.

Planeshifter replied:
Fixed.
